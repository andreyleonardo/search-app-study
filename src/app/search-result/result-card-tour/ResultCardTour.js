import React from 'react';
import he from 'he';
import PropTypes from 'prop-types';
import './ResultCardTour.scss';

const ResultCardTour = ({
  title, rating, currency, price, isSpecialOffer, onClick, tabIndex
}) => (
  <div
    className="result_card_tour"
    role="button"
    onKeyPress={onClick}
    onClick={onClick}
    tabIndex={tabIndex}
  >
    <p>{he.decode(title)}</p>
    <p>{rating}</p>
    <p>{currency + price}</p>
    <p>{isSpecialOffer && 'Special Offer'}</p>
  </div>
);

ResultCardTour.propTypes = {
  title: PropTypes.string.isRequired,
  rating: PropTypes.string,
  currency: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  isSpecialOffer: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  tabIndex: PropTypes.number.isRequired
};

ResultCardTour.defaultProps = {
  rating: '0',
  isSpecialOffer: false
};

export default ResultCardTour;
