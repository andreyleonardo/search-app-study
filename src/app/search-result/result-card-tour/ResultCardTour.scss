.result_card_tour {
  cursor: pointer;
  margin: 3px;
  padding: 0.2em;
  background-color: white;
  min-height: 100px;
  transition: box-shadow 0.5s;
  &:hover {
    box-shadow: 0px 0px 2px 5px rgba(255,255,255,1);
  }
}
