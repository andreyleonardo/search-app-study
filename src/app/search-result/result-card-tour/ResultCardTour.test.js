import React from 'react';
import ResultCardTour from './ResultCardTour';

describe('Result Card Tour', () => {
  let resultCardTour;
  let props;

  beforeEach(() => {
    props = {
      title: 'Tour Title',
      currency: '$',
      price: '30',
      rating: '4.8',
      onClick: jest.fn(),
      tabIndex: 1
    };

    resultCardTour = shallow(<ResultCardTour {...props} />);
  });

  it('renders without crashing', () => {
    expect(resultCardTour).toMatchSnapshot();
  });

  // it('renders Special Offer text when isSpecialOffer is true', () => {
  //   props = {
  //     ...props,
  //     isSpecialOffer: true
  //   };
  //   resultCardTour = shallow(<ResultCardTour {...props} />);
  //
  //   expect(resultCardTour).toMatchSnapshot();
  // });
});
