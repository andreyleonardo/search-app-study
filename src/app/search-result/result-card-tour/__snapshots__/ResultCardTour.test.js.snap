// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Card Tour renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultCardTour
    currency="$"
    isSpecialOffer={false}
    onClick={[MockFunction]}
    price="30"
    rating="4.8"
    tabIndex={1}
    title="Tour Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Tour Title
        </p>,
        <p>
          4.8
        </p>,
        <p>
          $30
        </p>,
        <p />,
      ],
      "className": "result_card_tour",
      "onClick": [MockFunction],
      "onKeyPress": [MockFunction],
      "role": "button",
      "tabIndex": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tour Title",
        },
        "ref": null,
        "rendered": "Tour Title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "4.8",
        },
        "ref": null,
        "rendered": "4.8",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "$30",
        },
        "ref": null,
        "rendered": "$30",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
        },
        "ref": null,
        "rendered": false,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Tour Title
          </p>,
          <p>
            4.8
          </p>,
          <p>
            $30
          </p>,
          <p />,
        ],
        "className": "result_card_tour",
        "onClick": [MockFunction],
        "onKeyPress": [MockFunction],
        "role": "button",
        "tabIndex": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tour Title",
          },
          "ref": null,
          "rendered": "Tour Title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4.8",
          },
          "ref": null,
          "rendered": "4.8",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$30",
          },
          "ref": null,
          "rendered": "$30",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
          },
          "ref": null,
          "rendered": false,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
